pkg/virt-launcher/virtwrap/network/network.go

```diff
@@ -0,0 +1,250 @@
+	// Change eth0 MAC
+	// Create macvlan and set fake address
+	// remove eth0 IP
+	// Start DHCP
+
+	nic := &VIF{Name: podInterface}
+	link, err := netlink.LinkByName(podInterface)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get a link for interface: %s", podInterface)
+		return err
+	}
+
+	// get IP address
+	addrList, err := netlink.AddrList(link, netlink.FAMILY_V4)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get an ip address for %s", podInterface)
+		return err
+	}
+	if len(addrList) == 0 {
+		return fmt.Errorf("No IP address found on %s", podInterface)
+	}
+	nic.IP = addrList[0]
+
+	// Get interface gateway
+	routes, err := netlink.RouteList(link, netlink.FAMILY_V4)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get routes for %s", podInterface)
+		return err
+	}
+	if len(routes) == 0 {
+		return fmt.Errorf("No gateway address found in routes for %s", podInterface)
+	}
+	nic.Gateway = routes[0].Gw
+
+	// Get interface MAC address
+	mac, err := GetMacDetails(podInterface)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get MAC for %s", podInterface)
+		return err
+	}
+	nic.MAC = mac
+
+	// Remove IP from POD interface
+	err = netlink.AddrDel(link, &nic.IP)
+
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to delete link for interface: %s", podInterface)
+		return err
+	}
+
+	// Set interface link to down to change its MAC address
+	err = netlink.LinkSetDown(link)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to bring link down for interface: %s", podInterface)
+		return err
+	}
+
+	_, err = ChangeMacAddr(podInterface)
+	if err != nil {
+		return err
+	}
+
+	err = netlink.LinkSetUp(link)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to bring link up for interface: %s", podInterface)
+		return err
+	}
+
+	// Create a macvlan link
+	macvlan := &netlink.Macvlan{
+		LinkAttrs: netlink.LinkAttrs{
+			Name:        macVlanIfaceName,
+			ParentIndex: link.Attrs().Index,
+		},
+		Mode: netlink.MACVLAN_MODE_BRIDGE,
+	}
+
+	//Create macvlan interface
+	if err := netlink.LinkAdd(macvlan); err != nil {
+		log.Log.Reason(err).Errorf("failed to create macvlan interface")
+		return err
+	}
+
+	//get macvlan link
+	macvlink, err := netlink.LinkByName(macVlanIfaceName)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get a link for interface: %s", macVlanIfaceName)
+		return err
+	}
+	err = netlink.LinkSetUp(macvlink)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to bring link up for interface: %s", macVlanIfaceName)
+		return err
+	}
+
+	// set fake ip on macvlan interface
+	fakeaddr, err := netlink.ParseAddr(macVlanFakeIP)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to bring link up for interface: %s", macVlanIfaceName)
+		return err
+	}
+
+	if err := netlink.AddrAdd(macvlink, fakeaddr); err != nil {
+		log.Log.Reason(err).Errorf("failed to set macvlan IP")
+		return err
+	}
+
+	// Start DHCP
+	go func() {
+		dhcp.SingleClientDHCPServer(
```

Comment:
I am missing error handling here. We at least want to log something, maybe even stop the whole pod?

Reply:
I still see no error handling in the latest code change. Will you address that?